core/src/main/java/io/questdb/griffin/model/RuntimeIntervalModelBuilder.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.model;
+
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.SqlException;
+import io.questdb.std.LongList;
+import io.questdb.std.Mutable;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+
+/**
+ * Collects interval during query parsing
+ * and records them as 2 types:
+ * - static list of intervals as 2 long points [lo, hi] in staticIntervals list
+ * - dynamic list of functions.
+ * <p>
+ * When first interval involving function is added all data starts to be encoded in 4 longs in staticPeriods
+ * 0: lo (long)
+ * 1: hi (long)
+ * 2: operation (short), period type (short), adjustment (int)
+ * 3: period (int), count (int)
+ * <p>
+ * and the index when it happens stored in pureStaticCount
+ */
+public class RuntimeIntervalModelBuilder implements Mutable {
+    public static final long DYNAMIC_LO_HI = Numbers.INT_NaN + 2;
```

Comment:
interesting choice of constant :) perhaps `Numbers.LONG_NaN` ?

Reply:
I removed it, added another short flag to indicate which side of interval is the function